cohorts/hash.py

```diff
@@ -0,0 +1,42 @@
+import copy
+
+# courtesy of https://stackoverflow.com/a/8714242/3457743
```

Comment:
Can you throw this in the docstring and move the function to utils or something? Extra file seems unnecessary to me; but if you disagree I don't feel strongly.